floodsub.go

```diff
@@ -303,59 +350,47 @@ func (p *PubSub) publishMessage(from peer.ID, msg *pb.Message) error {
 	return nil
 }
 
-type addSub struct {
+type addSubReq struct {
 	topic string
-	sub   bool
-	resp  chan chan *Message
+	resp  chan *Subscription
 }
 
-func (p *PubSub) Subscribe(ctx context.Context, topic string) (<-chan *Message, error) {
-	return p.SubscribeComplicated(&pb.TopicDescriptor{
-		Name: proto.String(topic),
-	})
-}
-
-type topicReq struct {
-	resp chan []string
-}
+// Subscribe returns a new Subscription for the given topic
+func (p *PubSub) Subscribe(topic string) *Subscription {
```

Comment:
We should ensure that all subscribe calls go through `SubscribeByTopicDescriptor`, it will make things easier on us moving forward